vendor/views/vendor.py

```diff
@@ -77,14 +77,14 @@ def get(self, request, *args, **kwargs):
 
             return render(request, self.template_name, context)
 
-        profile, created = self.request.user.customer_profile.get_or_create(site=set_default_site_id())
+        profile, created = self.request.user.customer_profile.get_or_create(site=request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(request)`